src/mixins/atoms.js

```diff
@@ -71,18 +77,111 @@ export const AtomsMixin = (superclass) => class extends superclass {
+                const element2 = basis.getElementByIndex(index2);
+                if (index2 <= index1 || !this.areElementsBonded(element1, coordinate1, element2, coordinate2)) return;
+                const bond = this.getBondObjectAsCylinder(element1, index1, coordinate1, element2, index2, coordinate2);
+                this.bondsGroup.add(bond);
+            });
+        });
+        this.structureGroup.add(this.bondsGroup);
+    }
+
+    cylindricalSegment(A, B, radius, material) {
+        var vec = B.clone();
+        vec.sub(A);
+        var h = vec.length();
+        vec.normalize();
+        var quaternion = new THREE.Quaternion();
+        quaternion.setFromUnitVectors(new THREE.Vector3(0, 1, 0), vec);
+        var geometry = new THREE.CylinderGeometry(radius, radius, h, 32);
+        geometry.translate(0, h / 2, 0);
+        var cylinder = new THREE.Mesh(geometry, material);
+        cylinder.applyQuaternion(quaternion);
+        cylinder.position.set(A.x, A.y, A.z);
+        return cylinder;
+    }
+
+    /**
+     * Draw bond with cylinder geometry.
+     */
+    getBondObjectAsCylinder(element1, index1, coordinate1, element2, index2, coordinate2) {
+        const vector1 = new THREE.Vector3(...coordinate1.value);
+        const vector2 = new THREE.Vector3(...coordinate2.value);
+        const direction = new THREE.Vector3().subVectors(vector2, vector1);
+        const height = direction.length();
+        direction.normalize();
+        const quaternion = new THREE.Quaternion();
+        quaternion.setFromUnitVectors(new THREE.Vector3(0, 1, 0), direction);
+        const geometry = new THREE.CylinderGeometry(0.1, 0.1, height, 8, 1);
+        geometry.translate(0, height / 2, 0);
+        const material = new THREE.MeshBasicMaterial();
+        const bond = new THREE.Mesh(geometry, material);
+        bond.applyQuaternion(quaternion);
+        bond.position.set(vector1.x, vector1.y, vector1.z);
+        bond.name = `${element1}-${index1}:${element2}-${index2}`;
+        return bond;
+    }
+
+    /**
+     * Draw bond with line geometry.
+     */
+    getBondObjectAsLine(element1, index1, coordinate1, element2, index2, coordinate2) {
+        const material = new THREE.LineBasicMaterial();
+        const geometry = new THREE.Geometry();
+        geometry.vertices.push(new THREE.Vector3(...coordinate1.value));
+        geometry.vertices.push(new THREE.Vector3(...coordinate2.value));
+        const bond = new THREE.Line(geometry, material);
+        bond.name = `${element1}-${index1}:${element2}-${index2}`;
+        return bond;
+    }
+
+    /**
+     * Remove bonds.
+     */
+    removeBonds() {
+        this.structureGroup.remove(this.bondsGroup);
+    }
+
+    toggleBonds() {
+        if (this.areBondsDrawn) {
+            this.removeBonds();
+            this.areBondsDrawn = false;
+        } else {
+            this.addBonds();
+            this.areBondsDrawn = true;
+        }
+    }
+
```

Comment:
There's a fair amount of logic for bonds, I suggest moving above to a bonds mixin and calling initBonds above instead of lines 14-16 and 21